test/e2e/serial/tests/configuration.go

```diff
@@ -906,3 +786,21 @@ func accumulateKubeletConfigNames(cms []corev1.ConfigMap) sets.Set[string] {
 	}
 	return cmNames
 }
+
+func waitForMcpsCondition(cli client.Client, ctx context.Context, mcps []*machineconfigv1.MachineConfigPool, condition machineconfigv1.MachineConfigPoolConditionType) {
+	var wg sync.WaitGroup
+	for _, mcp := range mcps {
+		wg.Add(1)
+		klog.Infof("wait for mcp %q to start updating", mcp.Name)
+		go func(mcpool *machineconfigv1.MachineConfigPool) {
+			defer GinkgoRecover()
+			defer wg.Done()
+			err := wait.With(cli).
+				Interval(configuration.MachineConfigPoolUpdateInterval).
+				Timeout(configuration.MachineConfigPoolUpdateTimeout).
+				ForMachineConfigPoolCondition(ctx, mcpool, condition)
+			Expect(err).ToNot(HaveOccurred())
+		}(mcp)
+	}
+	wg.Wait()
+}
```

Comment:
thanks, this is nice. There's not a hard and fast rule about extracting code in test code, and I've been advocating for actually leaving copy/pasted code around (and I still am).
The reasons are mostly two:
- avoid subtle coupling: it's relatively common to have flows which are similar but different in tiny details. A helper function would need to have `if`s and flags. This tend to snowball overtime, leading to actually worse readability and maintainability. Rather than rush to extract helpers, usually better to step back and review the flow of code, and evaluate other solutions like tabular testing
- readability. Being able to follow the flow without jumping too much in/out functions is still a huge plus even in 2024 where AI-powered tools and powerful IDEs are a commodity.
The rule of thumb I'm following is when we have a really self-contained, well defined piece of code which embodies a check or a step which cannot really fail and/or than we heavily depend on, then it is a candidate to be factored out. Repetitiviness is also a factor: up to 2-3 times is OK

Reply:
Thanks for highlighting these useful points, I've updated the log to be more specific to the actual condition that this check is waiting for (line 794)